src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -346,6 +383,39 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
         return ResponseUtil.wrapOrNotFound(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises-with-participations/:id : get the "id" programmingExercise.
+     *
+     * @param id the id of the programmingExercise to retrieve
+     * @return the ResponseEntity with status 200 (OK) and with body the programmingExercise, or with status 404 (Not Found)
+     */
+    @GetMapping("/programming-exercises-with-participations/{id}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithAllParticipations(@PathVariable Long id) {
```

Comment:
rename, it's confusing because I would assume that you load all student participations here

Reply:
You are right, I renamed it.